3711/1/ruby_gem.rb

```diff
@@ -0,0 +1,37 @@
+class RubyGem
+  attr_reader :score, :gem_data
+
+  ATTR_WEIGTHS = {
+    used_by: 10,
+    watched_by: 10,
+    stars: 10,
+    forks: 10,
+    contributors: 10,
+    issues: 10
+  }.freeze
+
+  def initialize(gem_data)
+    @gem_data = gem_data
+    @score = score_rating
+  end
+
+  def to_s
+    [
+      @gem_data[:name],
+      "used by #{@gem_data[:used_by]}",
+      "watched by #{@gem_data[:watched_by]}",
+      "#{@gem_data[:stars]} stars",
+      "#{@gem_data[:forks]} forks",
+      "#{@gem_data[:contributors]} contributors",
+      "#{@gem_data[:issues]} issues"
+    ].join("\t| ")
+  end
+
+  private
+
+  def score_rating
+    ATTR_WEIGTHS.each_pair.map do |key, value|
```

Comment:
Currently all weights are same and it's possible to create one constant equal 10.
If you really want to have this hash, maybe this weights should be stored in some config file.
Or just let weights be different in this hash